### ISSUE TITLE
unwanted LCC class 36 appears in final `ecoregionGroups` when it should be converted

### ISSUE THREAD

**Issue by CeresBarros** [PredictiveEcology/Biomass_borealDataPrep#49]
Two parameters govern how Biomass_borealDataPrep should treat different land cover classes:
`forestedLCCClasses` determines which classes will be kept in the simulation (and treated as forest). it to `c(1:15, 20, 32, 34:35)`
`LCCClassesToReplaceNN ` determines which of the classes in `forestedLCCClasses` should be converted to forest classes and defaults to `34:35` (LCC2005 burns)

The sequence of operations that takes care of this is:
1. `forestedLCCClasses `determines which classes are kept in the map and which are "masked"
in `Biomass_boreal*` these lines convert classes NOT in `forestedLCCClasses`  to NAs, so they don't enter the initial `ecoregionGroups`
```
lccPixelsRemoveTF <- !(sim$rstLCC[] %in% P(sim)$forestedLCCClasses)
pixelsToRm <- lccPixelsRemoveTF | pixelsToRm
(...)
rstLCCAdj[pixelsToRm] <- NA
```
2. then `LCCClassesToReplaceNN` determines which of the classes  in `forestedLCCClasses`   are supposed to be replaced by forested classes
in  `LandR::convertUnwantedLCC` this line searches for the `LCCClassesToReplaceNN`  (= `classesToReplace`) within the initial `ecoregionGroups`
```
rstUnwantedLCC[gsub(".*_", "", ecoregionGroupVec) %in% classesToReplace] <- 1
```

Hence, to convert a class 36 (urban in LCC2005) to a forest class I have included in both parameters, just like 34:35.

However, in my simulations I have found that class 36 (but not 34/35, which also existed as far as I remember) is present in my final `ecoregionGroups`. Somewhere, this class is being introduced or not fully converted.

**Comment by achubaty** [comment]
NOTE: LandWeb puts 36 only in `LCCClassesToReplaceNN` (i.e., not in both params) and does not end up with 36 in `ecoregionGroups`.

**Comment by CeresBarros** [comment]
Yes, because LandWeb is masking class 36 (converting it to NA) on step 1. So step 2 is not really doing anything to the pixels that had class 36, because that class shouldn't appear on the initial `ecoregionGroups` (i.e. `ecoregionGroupVec`)

**Comment by achubaty** [comment]
@CeresBarros, has this all been resolved in #50?

**Comment by CeresBarros** [comment]
Yes. Closing.
